### PR TITLE
Validate task payload before creation

### LINKED ISSUE #5577: UI: Create task view should provide options or templates for generic-worker/docker-worker, and should be able to lint payload before submission
Currently if you try to create a payload it would only fail when it will reach the worker.
Would be good to provide templates in "Create task" view (because docker worker and generic worker have different payloads), plus when desired template/type is selected, we should be able to easily validate it against schema before it will reach queue and worker

Also, `generic-worker` binary has `show-payload-schema`, it would be good to add `validate-payload` or `lint-payload` so one can do it from CLI also

### PR DESCRIPTION
**Pull request by lotas**
* Docker compose provides correct defaults in task view.
* Adds functionality to validate task payload in UI.
* "Reset editor" takes currently selected schema in consideration and produces valid paylaod

Task schema validation is done on the UI only and with the help of the following schemas:
* [task schema](https://docs.taskcluster.net/docs/reference/platform/queue/task-schema)
* worker type, selected by user:
  * [docker worker payload](https://docs.taskcluster.net/docs/reference/workers/docker-worker/payload)
  * [generic worker simple posix](https://docs.taskcluster.net/docs/reference/workers/generic-worker/simple-posix-payload)
  * [generic worker multiuser posix](https://docs.taskcluster.net/docs/reference/workers/generic-worker/multiuser-posix-payload)
  * [generic worker multiuser windows](https://docs.taskcluster.net/docs/reference/workers/generic-worker/multiuser-windows-payload)
  * [generic worker docker posix](https://docs.taskcluster.net/docs/reference/workers/generic-worker/docker-posix-payload)

Fixes #5577 

image

image

Validation is done manually by selecting worker schema and clicking `Validate schema` button

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -97,6 +97,7 @@ services:
       GRAPHQL_SUBSCRIPTION_ENDPOINT: http://taskcluster/graphql
       GRAPHQL_ENDPOINT: http://taskcluster/graphql
       UI_LOGIN_STRATEGY_NAMES: local
+      SITE_SPECIFIC: '{"tutorial_worker_pool_id":"docker-compose/generic-worker","tutorial_worker_schema":"generic-simple-posix"}'
```
Comment (lotas) [inline docker-compose.yml:100]: this helps to set initial worker pool id properly in the create task view

File: docker-compose.yml [hunk 2]
```diff
@@ -183,7 +184,7 @@ services:
       TRUST_PROXY: 'true'
       READ_DB_URL: postgresql://postgres@postgres:5432/taskcluster
       WRITE_DB_URL: postgresql://postgres@postgres:5432/taskcluster
-      DB_CRYPTO_KEYS: '[{"id": "dev-init", "algo": "aes-256", "key": "AUZzegzU1Xp3dW2tPRU615HXI04oJTt9NDIokH3HXN4="}]'
+      DB_CRYPTO_KEYS: '[{"id":"dev-init","algo":"aes-256","key":"AUZzegzU1Xp3dW2tPRU615HXI04oJTt9NDIokH3HXN4="}]'
```
Comment (lotas) [inline docker-compose.yml:187]: just the whitespace, switched to `JSON.stringify({})`

File: ui/src/utils/validateTaskPayloadSchemas.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+const schemas = {};
+const fetchSchema = async (service, schema) => {
+  const res = await fetch(urls.schema(service, schema));
+
+  return res.json();
+};
+
+const prefetch = async () => {
+  ajv.addSchema(await fetchSchema('common', 'metaschema.json'));
```
Comment (lotas) [inline ui/src/utils/validateTaskPayloadSchemas.js:12]: those three are needed to validate task schema

File: ui/src/utils/validateTaskPayloadSchemas.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  ajv.addSchema(await fetchSchema('queue', 'v1/task.json'));
+  ajv.addSchema(await fetchSchema('queue', 'v1/task-metadata.json'));
+};
+
+const prefetchPromise = prefetch();
+
+export const formatErrorDetails = error => {
+  const msg = [error.message];
+
+  if (error.keyword === 'type') {
```
Comment (lotas) [inline ui/src/utils/validateTaskPayloadSchemas.js:22]: other types are quite self-explanatory in message, no need to add extra messages

**Review by matt-boris — APPROVED** [review]
👏🏻

**Pull request merged**